daemon/src/command.rs

```diff
@@ -34,14 +34,66 @@ impl Executor {
             .await
             .context("Failed to load CFD")?;
 
-        let event = command(cfd).context("Failed to execute command on CFD")?;
+        let return_val = command(cfd).context("Failed to execute command on CFD")?;
 
-        self.process_manager
-            .send(process_manager::Event::new(event))
-            .await
-            .context("ProcessManager is disconnected")?
-            .context("Failed to process new domain event")?;
+        let (event, rest) = return_val.extract_event();
+
+        if let Some(event) = event {
+            self.process_manager
+                .send(process_manager::Event::new(event))
+                .await
+                .context("ProcessManager is disconnected")?
+                .context("Failed to process new domain event")?;
+        }
+
+        Ok(rest)
+    }
+}
+
+// TODO: Delete this weird thing once all our commands return only an `Event` and not other stuff as
+// well.
+pub trait ExtractEventFromTuple {
```

Comment:
wow. ingenious effort. It's weird, but at least it puts all the weirdness in a single spot!

Reply:
Figured that that would impress you :)